pyhap/accessory_driver.py

```diff
@@ -859,118 +861,99 @@ def set_characteristics(self, chars_query, client_addr):
+        expired = False
+        pid = chars_query.get(HAP_REPR_PID, None)
+        if pid is not None:
+            expire_time = self.prepared_writes.get(client_addr, {}).pop(pid, None)
+            expired = expire_time is None or time.time() > expire_time
 
-            if HAP_PERMISSION_NOTIFY in cq:
-                char_topic = get_topic(aid, iid)
-                action = "Subscribed" if cq[HAP_PERMISSION_NOTIFY] else "Unsubscribed"
-                logger.debug(
-                    "%s client %s to topic %s", action, client_addr, char_topic
-                )
-                self.async_subscribe_client_topic(
-                    client_addr, char_topic, cq[HAP_PERMISSION_NOTIFY]
-                )
+        primary_accessory = self.accessory
+        primary_aid = primary_accessory.aid
 
-            if HAP_REPR_VALUE not in cq:
+        for query in queries:
+            if HAP_REPR_VALUE not in query and not expired:
                 continue
 
-            updates.setdefault(aid, {})[iid] = cq[HAP_REPR_VALUE]
+            aid, iid = query[HAP_REPR_AID], query[HAP_REPR_IID]
+            value = query.get(HAP_REPR_VALUE, None)
+            write_response_requested = query.get(HAP_REPR_WRITE_RESPONSE, False)
 
-        for aid, new_iid_values in updates.items():
-            if self.accessory.aid == aid:
-                acc = self.accessory
+            if aid == primary_aid:
+                acc = primary_accessory
             else:
                 acc = self.accessory.accessories.get(aid)
+            char = acc.get_characteristic(aid, iid)
 
-            updates_by_service = {}
-            char_to_iid = {}
-            for iid, value in new_iid_values.items():
-                # Characteristic level setter callbacks
-                char = acc.get_characteristic(aid, iid)
-
-                set_result, set_result_value = _wrap_char_setter(char, value, client_addr)
-                if set_result != HAP_SERVER_STATUS.SUCCESS:
-                    had_error = True
-
-                setter_results[aid][iid] = set_result
-
-                if set_result_value is not None:
-                    if setter_responses.get(aid, None) is None:
-                        logger.warning(
-                            "Returning write response '%s' when it wasn't requested for %s %s",
-                            set_result_value, aid, iid
-                        )
-                    had_write_response = True
-                    setter_responses.setdefault(aid, {})[iid] = set_result_value
-
-                if not char.service or (
-                    not acc.setter_callback and not char.service.setter_callback
-                ):
-                    continue
-                char_to_iid[char] = iid
-                updates_by_service.setdefault(char.service, {}).update({char: value})
+            set_result, set_result_value = (
+                HAP_SERVER_STATUS.INVALID_VALUE_IN_REQUEST,
+                None,
+            )
```

Comment:
```suggestion
            set_result = HAP_SERVER_STATUS.INVALID_VALUE_IN_REQUEST
            set_result_value = None
```

Reply:
This was done by black

Reply:
Maybe there was a comma out of place? I can't replicate that

```
bdraco@MacBook-Pro-9 HAP-python % black pyhap/accessory_driver.py
All done! ✨ 🍰 ✨
1 file left unchanged.
bdraco@MacBook-Pro-9 HAP-python % git diff -w
diff --git a/pyhap/accessory_driver.py b/pyhap/accessory_driver.py
index 7bb0984..6cc6966 100644
--- a/pyhap/accessory_driver.py
+++ b/pyhap/accessory_driver.py
@@ -893,10 +893,9 @@ class AccessoryDriver:
                 acc = self.accessory.accessories.get(aid)
             char = acc.get_characteristic(aid, iid)
 
-            set_result, set_result_value = (
-                HAP_SERVER_STATUS.INVALID_VALUE_IN_REQUEST,
-                None,
-            )
+            set_result = HAP_SERVER_STATUS.INVALID_VALUE_IN_REQUEST
+            set_result_value = None
+
             if value is not None:
                 set_result, set_result_value = _wrap_char_setter(
                     char, value, client_addr
@@ -906,12 +905,10 @@ class AccessoryDriver:
                 set_result_value is not None and write_response_requested
             )
 
-            aid_results = results[aid]
-
-            aid_results[iid] = {HAP_REPR_STATUS: set_result}
+            aid_iid_results = {HAP_REPR_STATUS: set_result}
             if return_value_in_response:
-                aid_results[iid][HAP_REPR_VALUE] = set_result_value
-
+                aid_iid_results[HAP_REPR_VALUE] = set_result_value
+            results[aid][iid] = aid_iid_results
             char_to_iid[char] = iid
             service = char.service
 
```

Reply:
Sorry, I've misread this code as having both inits in the same line. Doing it on separate - ok.